tests/unit/parsers/test.manifestjson.js

```diff
@@ -874,10 +936,47 @@ describe('ManifestJSONParser', () => {
 
       expect(manifestJSONParser.isValid).toEqual(true);
       const { warnings } = addonLinter.collector;
-      expect(warnings[0].code).toEqual(messages.MANIFEST_CSP_UNSAFE_EVAL.code);
+      expect(warnings[0].code).toEqual(messages.MANIFEST_CSP_UNSAFE_EVAL);
       expect(warnings[0].message).toEqual(
-        "Using 'eval' has strong security and performance implications."
+        messages.manifestCspUnsafeEval('content_security_policy').message
+      );
+
+      // Clear any warnings and errors collected.
+      addonLinter.collector.warnings = [];
+      addonLinter.collector.errors = [];
+
+      // Manifest v3 formats.
+      const contentSecurityPolicy = {
+        extension_pages: invalidValue,
+        content_scripts: invalidValue,
+        // Alias for content_scripts.
+        isolated_world: invalidValue,
+      };
+
+      const jsonV3 = validManifestJSON({
+        content_security_policy: contentSecurityPolicy,
+        applications: {
+          // The new content_security_policy syntax is only supported
+          // on Firefox >= 72.
+          gecko: { strict_min_version: '72.0' },
+        },
+      });
+
+      const manifestV3JSONParser = new ManifestJSONParser(
+        jsonV3,
+        addonLinter.collector
       );
+
+      expect(manifestV3JSONParser.isValid).toEqual(true);
+      const warningsV3 = addonLinter.collector.warnings;
+
+      const keys = Object.keys(contentSecurityPolicy);
```

Comment:
Same here, can we add an assertion that makes sure that we have exactly two warnings that match the expected csp warnings?

Reply:
As mentioned in the other comment, I can add an assertion to be sure that we got the expected number of warnings (which is 3 for this test)